src/main/java/ru/sbt/mipt/oop/HallEventHandler.java

```diff
@@ -0,0 +1,46 @@
+package ru.sbt.mipt.oop;
+
+import static ru.sbt.mipt.oop.SensorEventType.DOOR_CLOSED;
+
+public class HallEventHandler implements SensorEventHandler {
+	private final CommandSender commandSender;
+
+	public HallEventHandler(CommandSender commandSender) {
+		this.commandSender = commandSender;
+	}
+
+	@Override
+	public void handle(SmartHome smartHome, SensorEvent event) {
+		if (event.getType() == DOOR_CLOSED) {
+			smartHome.execute(object -> {
+				if (!(object instanceof Room)) {
+					return;
+				}
+				Room room = (Room) object;
+				if (room.getName().equals("hall")) {
+					room.execute(component -> {
+						if (!(component instanceof Door)) {
+							return;
+						}
+						Door door = (Door) component;
+						if (door.getId().equals(event.getObjectId())) {
+							setOffAllLight(smartHome);
+						}
+					});
+				}
+			});
+		}
+	}
+
+	private void setOffAllLight(SmartHome smartHome) {
+		smartHome.execute(object -> {
+			if (!(object instanceof Light)) {
+				return;
+			}
+			Light light = (Light) object;
+			light.setOn(false);
+			SensorCommand command = new SensorCommand(CommandType.LIGHT_OFF, light.getId());
+			new CommandSender().sendCommand(command);
```

Comment:
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D0%BF%D0%B5%D1%80%D0%B5%D0%B4%D0%B0%D0%B2%D0%B0%D0%B9%D1%82%D0%B5-%D0%B7%D0%B0%D0%B2%D0%B8%D1%81%D0%B8%D0%BC%D0%BE%D1%81%D1%82%D0%B8-%D1%81%D0%BD%D0%B0%D1%80%D1%83%D0%B6%D0%B8